samples/SocialSample/Startup.cs

```diff
@@ -130,48 +131,45 @@ 127.0.0.1 MsSecSample.localhost.this
                 options.TokenEndpoint = "https://github.com/login/oauth/access_token";
                 options.UserInformationEndpoint = "https://api.github.com/user";
                 options.ClaimsIssuer = "OAuth2-Github";
+                options.SaveTokensAsClaims = false;
                 // Retrieving user information is unique to each provider.
                 options.Notifications = new OAuthAuthenticationNotifications()
                 {
-                    OnGetUserInformationAsync = async (context) =>
+                    OnAuthenticated = async notification =>
                     {
                         // Get the GitHub user
-                        var userRequest = new HttpRequestMessage(HttpMethod.Get, context.Options.UserInformationEndpoint);
-                        userRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", context.AccessToken);
+                        var client = new HttpClient(notification.Options.BackchannelHttpHandler);
+                        var userRequest = new HttpRequestMessage(HttpMethod.Get, notification.Options.UserInformationEndpoint);
+                        userRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", notification.AccessToken);
                         userRequest.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-                        var userResponse = await context.Backchannel.SendAsync(userRequest, context.HttpContext.RequestAborted);
+                        var userResponse = await client.SendAsync(userRequest, notification.HttpContext.RequestAborted);
                         userResponse.EnsureSuccessStatusCode();
                         var text = await userResponse.Content.ReadAsStringAsync();
                         var user = JObject.Parse(text);
 
-                        var identity = new ClaimsIdentity(
-                            context.Options.AuthenticationScheme,
-                            ClaimsIdentity.DefaultNameClaimType,
-                            ClaimsIdentity.DefaultRoleClaimType);
+                        var identity = (ClaimsIdentity) notification.Principal.Identity;
```

Comment:
@blowdart yup, I clearly prefer having a single property, even if we need to have a ugly cast.